src/views/iam/RolePermissionTab.vue

```diff
@@ -16,37 +16,404 @@
 // under the License.
 
 <template>
-  <div>
-    Boilerplate for implementation
-    {{ loading }}
-    {{ resource }}
+  <a-icon v-if="loadingTable" type="loading" class="main-loading-spinner"></a-icon>
+  <div v-else>
+    <div v-if="updateTable" class="loading-overlay">
+      <a-icon type="loading" />
+    </div>
+    <div
+      class="rules-list ant-list ant-list-bordered"
+      :class="{'rules-list--overflow-hidden' : updateTable}" >
+
+      <div class="rules-table-item ant-list-item">
+        <div class="rules-table__col rules-table__col--grab"></div>
+        <div class="rules-table__col rules-table__col--rule rules-table__col--new">
+          <a-p-i-auto-complete
+            @selectedRecord="onRuleSelect"
+            :error="newRuleSelectError"
+            :data="$store.getters.apis"
```

Comment:
@RitchieVincent the api dropdown is very slow, can you instead create a variable in `data` and in `mounted` that the variable (i.e. api array data source) using `Object.keys($store.getters.apis)`?

Reply:
Also if we can sort the API names list in ascending order.

Reply:
This has been completed as you requested now. However it's still a little slow, but I'm not sure what else can be done about it. It's needing to load 500+ list items, so it's understandable that it'll be slightly slow.